analysis/03_new_model_training/02_model_training-baselines.md

```diff
@@ -0,0 +1,216 @@
+# Model training
+
+Baseline is the simplest algorithm that provides predictions
+without complex computations.
+For regression tasks, the Baseline returns the average
+of the target from training data.
+
+```python
+%load_ext jupyter_black
+```
+
+```python
+import statistics
+
+from sklearn import preprocessing
+from sklearn.preprocessing import RobustScaler
+from sklearn.model_selection import train_test_split
+from sklearn.dummy import DummyRegressor
+from sklearn.metrics import mean_squared_error
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+
+from utils import get_training_dataset
+```
+
+```python
+# Read csv file and import to df
+df = get_training_dataset()
+```
+
+```python
+# Show histogram of damage
+df.hist(column="percent_buildings_damaged", figsize=(4, 3))
+```
+
+```python
+# Hist plot after data stratification
+bins2 = [0, 0.00009, 1, 10, 50, 101]
+samples_per_bin2, binsP2 = np.histogram(
+    df["percent_buildings_damaged"], bins=bins2
+)
+plt.figure(figsize=(4, 3))
+plt.xlabel("Damage Values")
+plt.ylabel("Frequency")
+plt.plot(binsP2[1:], samples_per_bin2)
+```
+
+```python
+# Remove zeros from wind_speed
+df = df[(df[["wind_speed"]] != 0).any(axis=1)]
+
+df = df.drop(columns=["grid_point_id", "typhoon_year"])
+df
+```
+
+```python
+# Hist plot after removing rows where windspeed is 0
+bins2 = [0, 0.00009, 1, 10, 50, 101]
+samples_per_bin2, binsP2 = np.histogram(
+    df["percent_buildings_damaged"], bins=bins2
+)
+plt.figure(figsize=(4, 3))
+plt.xlabel("Damage Values")
+plt.ylabel("Frequency")
+plt.plot(binsP2[1:], samples_per_bin2)
+```
+
+```python
+print(samples_per_bin2)
+print(binsP2)
+```
+
+```python
+# Check the bins' intervalls
+# df["percent_buildings_damaged"].value_counts(bins=binsP2)
+```
+
+```python
+bin_index2 = np.digitize(df["percent_buildings_damaged"], bins=binsP2)
+```
+
+```python
+y_input_strat = bin_index2
+```
+
+```python
+# Split X and y from dataframe features
+
+features = ["wind_speed", "total_buildings"]
+X = df[features]
+# display(X.columns)
+y = df["percent_buildings_damaged"]
+
+scaler = preprocessing.StandardScaler().fit(X)
+X_scaled = scaler.transform(X)
+```
+
+```python
+# Define train_test_split
+X_train, X_test, y_train, y_test = train_test_split(
+    X_scaled,
+    df["percent_buildings_damaged"],
+    stratify=y_input_strat,
+    test_size=0.2,
+)
+```
+
+```python
+# create a dummy regressor
+dummy_reg = DummyRegressor(strategy="mean")
+
+# fit it on the training set
+dummy_reg.fit(X_train, y_train)
+
+bin_index_test = np.digitize(y_test, bins=binsP2)
+bin_index_train = np.digitize(y_train, bins=binsP2)
+
+# make predictions on the test set
+y_pred = dummy_reg.predict(X_test)
+y_pred_train = dummy_reg.predict(X_train)
+
+# Estimation of RMSE for train data per each bin
+mse_train_idx1 = mean_squared_error(
```

Comment:
All this repetition in the code and passing the index values in manually can be very prone to errors. Could you write a function to wrap this, or use a for loop at least?